.github/workflows/live-test.yml

```diff
@@ -1,24 +1,70 @@
 name: Live tests for B2K
 on:
-  push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
+  repository_dispatch:
+    types: [ok-to-test-command]
+permissions:
+  id-token: write
 jobs:
   liveTest:
+    env:
+      PR_NUMBER: ${{ github.event.client_payload.slash_command.args.named.PR_NUMBER }}
+      HEAD_REPO: ${{ github.event.client_payload.slash_command.args.named.HEAD_REPO }}
+      HEAD_BRANCH: ${{ github.event.client_payload.slash_command.args.named.HEAD_BRANCH }}
     strategy:
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     name: live test for b2k
     steps:
     - uses: actions/checkout@v2
+      # checkout PR head repo/branch
+      with:
+         repository: ${{ env.HEAD_REPO }}
+         ref: ${{ env.HEAD_BRANCH }}
+    - name: Login Azure Context
+      if: ${{ matrix.os != 'ubuntu-latest' }}
+      uses: Azure/login@v1
+      with:
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
```

Comment:
@cxznmhdcxz FYI - these secrets tenantID, SubscriptionID and CLIENT ID has to be added as GH secrets for this workflow to run smoothly.

Reply:
These are added to main repo secrets, thanks for the help @cxznmhdcxz .